nextcloudappstore/core/models.py

```diff
@@ -196,6 +202,58 @@ def _latest(self, releases):
             return None
 
 
+class AppRating(TranslatableModel):
+    app = ForeignKey('App', related_name='app', verbose_name=_('App'),
+                     on_delete=CASCADE)
+    user = ForeignKey(settings.AUTH_USER_MODEL, verbose_name=_('User'),
+                      on_delete=CASCADE, related_name='user')
+    rating = FloatField(verbose_name=_('Rating'), default=0.5,
+                        help_text=_('Rating from 0.0 (worst) to 1.0 (best)'))
+    rated_at = DateTimeField(auto_now=True, db_index=True)
+    translations = TranslatedFields(
+        comment=TextField(verbose_name=_('Rating comment'), default='',
+                          help_text=_('Rating comment in Markdown'))
+    )
+
+    class Meta:
+        unique_together = (('app', 'user'),)
+        verbose_name = _('App Rating')
+        verbose_name_plural = _('App Ratings')
+
+    def __str__(self) -> str:
+        return str(self.rating)
+
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+        # update rating on the app
+        app = self.app
+        day_range = settings.RATING_RECENT_DAY_RANGE
+        threshold = settings.RATING_THRESHOLD
+        app.rating_recent = self._compute_app_rating(app, day_range, threshold)
+        app.rating_overall = self._compute_app_rating(app, threshold=threshold)
+        app.save()
+
+    def _compute_app_rating(self, app: App, days: int = -1,
```

Comment:
Isn't `app` always the same as `self.app`? In other words: how come this method is basically static? And if we want it to be static, does it belong in a `Model` class?